### DISCUSSION TITLE
Why is the Text filter class removed

### DISCUSSION THREAD

**Discussion by ghost** [aiogram/aiogram#1227] [General]

**Comment by Olegt0rr** [comment]
All `Text()` functionality is already implemented and available in magic filters.
So framework author decided to remove redundant part. 
First it was kwargs-filters support, e.g. `text="hello"` (#978). 
Then – `Text()` (#1170).

That's why you should use [magic filters](https://docs.aiogram.dev/en/dev-3.x/dispatcher/filters/magic_filters.html) instead.

### Zen of Python:
```text
There should be one-- and preferably only one --obvious way to do it.
```

### +

I also agree with the point of ability to read all available methods in related filter. It's really handy for newbies. But for professional programmers it is more convenient to use the same behavior for the same data types:
- `F.text.startswith("hello")`
- `F.data.startswith("hello")`

**Reply by Olegt0rr** [reply]
Just updated my answer

**Reply by AIPACMan** [reply]
It seems [Olegt0rr](https://github.com/Olegt0rr) did not understand the question

**Reply by Olegt0rr** [reply]
My first answer was wrong, that's why I updated it 🤷‍♂️

**Comment by AIPACMan** [comment]
> All `Text()` functionality is already implemented and available in magic filters. So framework author decided to remove redundant part. First it was kwargs-filters support, e.g. `text="hello"` (#978). Then – `Text()` (#1170).
> 
> That's why you should use [magic filters](https://docs.aiogram.dev/en/dev-3.x/dispatcher/filters/magic_filters.html) instead.
> 
> ### Zen of Python:
> ```
> There should be one-- and preferably only one --obvious way to do it.
> ```
> 
> ### +
> I also agree with the point of ability to read all available methods in related filter. It's really handy for newbies. But for professional programmers it is more convenient to use the same behavior for the same data types:
> 
> * `F.text.startswith("hello")`
> * `F.data.startswith("hello")`

According to this logic, then it is already necessary to f.state instead of StateFilter, and f.command instead of Command and delete these classes, as well as delete "and_f", "or_f", "invert_f", which already have a replacement in the form of syntactic sugar python itself. You are trying to answer a question to which you do not know the answer, I assume that at least the one who removed the text filter knows why he did it. And what you are writing here is just a beautiful set of words that does not make sense and does not answer the question

**Reply by Olegt0rr** [reply]
1. Please stop to add new answers to the discussion. Use comments under the thread.
2. You can not share my opinion - it's your right. If you want to know the reason personally from the author, ask him personally. But according on the likes, the upvote and lack of comments – everything is about the same as I wrote.

**Reply by AIPACMan** [reply]
1. New answers? do you consider your unrelated copy-paste to be one of the answers?). By the same logic, my answer is also the answer to the higher question, why should it be in the comments?

**Reply by AIPACMan** [reply]
He liked your answer and I think he agree with your "answer"
If you don't know the answer, you don't answer. If you think your answer is at least partially correct, please answer at least one of the questions I asked and be ready to defend what you write

**Reply by Olegt0rr** [reply]
> 1. New answers? do you consider your unrelated copy-paste to be one of the answers?). By the same logic, my answer is also the answer to the higher question, why should it be in the comments?

Cause you're answering to my answer :)

**Reply by AIPACMan** [reply]
this is not an answer, you just copied the description of the commit)))
Yes, and so that you can delete all correspondence, as you have already done?
